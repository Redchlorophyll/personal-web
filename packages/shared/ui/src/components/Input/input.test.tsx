import { useState } from 'react';
import { Input } from './index';
import {
  renderWithProviders,
  screen,
  userEvent,
  act,
} from 'testing-library-react-custom';

function InputWrapper(props: { isDisabled?: boolean }) {
  const [value, setValue] = useState<string>('');

  const onChange = (event: React.FormEvent<HTMLInputElement>): void => {
    setValue((event.target as HTMLInputElement).value);
  };

  return (
    <div>
      <Input value={value} onChange={onChange} isDisabled={props.isDisabled} />
    </div>
  );
}

describe('shared - ui - Input', () => {
  test('should render properly', () => {
    const { baseElement } = renderWithProviders(<InputWrapper />);

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  test('It should render properly', () => {
    renderWithProviders(<InputWrapper />);
    const target = screen.getByLabelText('input');

    expect(target).toBeInTheDocument();
  });

  test('It should show right input when inputing value', async () => {
    const user = userEvent.setup();
    renderWithProviders(<InputWrapper />);
    const target = screen.getByLabelText('input');

    expect(screen.getByDisplayValue('')).toBeInTheDocument();
    await act(async () => await user.type(target, 'test'));
    expect(screen.getByDisplayValue('test')).toBeInTheDocument();
  });

  test('It should not allow text input when disabled', async () => {
    const user = userEvent.setup();
    renderWithProviders(<InputWrapper isDisabled={true} />);
    const target = screen.getByLabelText('input');

    expect(screen.getByDisplayValue('')).toBeInTheDocument();
    await act(async () => await user.type(target, 'test'));
    expect(screen.queryByDisplayValue('test')).not.toBeInTheDocument();
  });
});
