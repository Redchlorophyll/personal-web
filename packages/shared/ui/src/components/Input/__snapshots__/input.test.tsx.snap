// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared - ui - Input should render properly 1`] = `
<body>
  <div>
    <div>
      <div
        class="flex flex-col items-start"
      >
        <input
          aria-label="input"
          class="w-full h-[34px] rounded-[7px] border-[0.5px] border-solid px-[17px] py-[6px] dark:text-black-900 focus:dark:drop-shadow-[0px_1px_17px_#406FCB] outline-none disabled:bg-black-200 disabled:text-black-700 disabled:border-black-600 dark:disabled:bg-black-500 border-black-800 focus:border-primary-800"
          id=""
          name=""
          placeholder="Input Text Here..."
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</body>
`;
